### PR TITLE
Checkin generated manpage

### PR DESCRIPTION
**Pull request by TravisEz13**
# PR Summary

This pull request introduces several changes to the macOS packaging process, including updates to the build scripts and dependencies. The most significant changes involve adding a new packaging stage, removing the `ronn` dependency, and updating the man page generation process.

### macOS Packaging Process:
* Added a new `PackageMac` stage to the `.vsts-ci/mac.yml` file, which depends on the `BuildMac` stage and includes steps for macOS packaging using PowerShell. (`.vsts-ci/mac.yml`)

### Dependency Management:
* Removed the `ronn` dependency from the `Start-PSBootstrap` function in `build.psm1` and the `Get-PackageDependencies` function in `tools/packaging/packaging.psm1`. (`build.psm1`) [[1]](diffhunk://#diff-1eb6991ceacad71a638a32bcf0a4ec906c7ec150f5086732ab6d7702a624914aL2336-L2341) (`tools/packaging/packaging.psm1`) [[2]](diffhunk://#diff-8ac11d6f4bbedfce810550a3f8cfd0ee21f8b14085a5764d38577994c7e7c736L1639-R1639)

### Man Page Generation:
* Updated the man page generation process to use `roff` instead of `ronn` in the `New-ManGzip` function in `tools/packaging/packaging.psm1`. (`tools/packaging/packaging.psm1`)
* Added a new man page for `pwsh` in `assets/manpage/pwsh.1`. (`assets/manpage/pwsh.1`)

## PR Context

macOS man page generation is failing.  Checking in asset so we can get the release working.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
  - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
  - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
  - [x] None
  - **OR**
  - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.5/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
    - [ ] Experimental feature name(s): 
- **User-facing changes**
  - [x] Not Applicable
  - **OR**
  - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
  - [x] N/A or can only be tested interactively
  - **OR**
  - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
  - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
  - **OR**
  - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
    - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
      - [ ] Issue filed: 
    - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
      - [ ] Issue filed: 
    - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
      - [ ] Issue filed: 
    - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
      - [ ] Issue filed:

### CODE REVIEW

File: tools/packaging/packaging.psm1 [hunk 1]
```diff
@@ -1714,33 +1714,28 @@ function New-ManGzip
...
+        Copy-Item -Path $RoffFile -Destination $newRoffFile -Force
+        $RoffFile = $newRoffFile
     }
 
     # gzip in assets directory
     $GzipFile = "$RoffFile.gz"
     Write-Log "Creating man gz - running gzip..."
     Start-NativeExecution { gzip -f $RoffFile } -VerboseOutputOnError
 
+    if ($IsPreview.IsPresent)
```
Comment (adityapatwardhan) [inline tools/packaging/packaging.psm1:1734]: Why is this needed.. and only on preview?

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: tools/packaging/packaging.psm1 (hunk 1 above)
Comment (TravisEz13) [reply tools/packaging/packaging.psm1:1734]: we only create an extra file on preview.

**Review by adityapatwardhan — APPROVED** [review]

**Review by adityapatwardhan — APPROVED** [review]
